File: Kattis/Building Pyramids/BP.cpp
#include <iostream>
using namespace std;

int main(void) {

    //Initialization
    int numberOfBlocks;
    int totalPyramidLevels = 0;
    int sideSize = 1;
    int levelSize = 0;

    cin >> numberOfBlocks;

    while (numberOfBlocks >=0 ) {

        levelSize = sideSize * sideSize;

        if (numberOfBlocks >= levelSize) {

            numberOfBlocks -= levelSize;
            totalPyramidLevels += 1;
            sideSize +=2;

        } else {

            break;

        }

    }

    cout << totalPyramidLevels;

}

File: Kattis/Building Pyramids/BP.py
numberOfBlocks = int(input())
totalPyramidLevels = 0
sideSize = 1
levelSize = 0

while numberOfBlocks >= 0:
    levelSize = sideSize * sideSize
    if numberOfBlocks >= levelSize:
        numberOfBlocks -= levelSize
        totalPyramidLevels += 1
        sideSize += 2
    else:
        break

print(totalPyramidLevels)
